docs/development/pigs/pig-012.rst

```diff
@@ -0,0 +1,334 @@
+
+This high-level interface API is similar to what it is done in `Fermipy`_ or HAP in HESS, including
+also the options to save and recover session states, as well as serialization of intermediate data
+products and logging. It is flexible enough to allow the user to work with the API at any stage of the
+analysis process, and not only from the start to the very end or in automatized process.
+
+**Use cases**
+
+The use cases covered are in the scope of a single analysis and model, not parametrized variations in a 
+multidimensional grid space of variables, and within a single region (e.g. 10 deg region with 5 sources).
+
+The main use cases for analysis to be covered are:
+
+- 3D map analysis
+- 2D map analysis
+- 1D spectrum analysis
+
+Including the main methods for data reduction, modelling and fitting:
+
+- On vs on/off data reduction
+- Different background models
+- Joint vs stacked likelihood fitting
+- Diagnostics (residuals, significance, TS)
+- Spectral points
+
+Making a SED may be the final part of the analysis, as many SED methods require the full model and 
+all energy data.
+
+**Configuration file**
+
+The configuration file will be in YAML format exposing the parameters and values needed for each one 
+of the tasks involved in the analysis process. To generate the config file, we could add a 
+``gammapy analysis config`` command line tool which dumps the config file with all lines commented
+out, and the users can then uncomment and fill in the parameters and values they care about. As an 
+alternative users could copy & paste from a config file example in the docs. We will develop a 
+schema and validate / give good error messages on read.
+
+We roughly sketch below an example of a prototype configuration file in YAML format, just to illustrate
+how a structured schema could expose most of the parameter/values needed in a data analysis process.
+The configuration file should be explicit enough for the users to understand which parameters
+to edit in order to define a specific configuration for an analysis session or workflow. The final
+schema for this configuration file will be achieved iteratively during the development of the high-level
+interface and later on eventual improvements, also taking user feedback into account.
+
+Prototype configuration file.
+
+.. code::    
+    
+    analysis:
+        process:
+            # add options to allow either in-memory or disk-based processing
+            out_folder: "."  # default is current working directory
+            store_per_obs: {true, false}
+        reduce:
+            type: {"1D, "3D"}
+            stacked: {true, false}
+            background: {"irf", "reflected", "ring"}
+            roi: max_offset
+            exclusion: exclusion.fits
+        fit:
+            energy_min, energy_max
+        logging:
+            level: debug
+
+    grid:
+        spatial: center, width, binsz
+        energy: min, max, nbins
+        time: min, max
```

Comment:
The reader of this config file should be able to interpret the units. Should we mention it here?

Reply:
+1 to mention "Config file should use units where it makes sense, e.g. "angle: 3 deg" instead of "angle: 3".

Reply:
OK, done.